src/coreclr/src/nativeaot/System.Private.CoreLib/src/Internal/DeveloperExperience/DeveloperExperience.cs

```diff
@@ -36,10 +33,17 @@ private static bool IsMetadataStackTraceResolutionDisabled()
             return disableMetadata;
         }
 
-        public virtual void WriteLine(string s)
+        internal static void WriteStackTraceLine(string s)
```

Comment:
Nit: 
If we want this to be generic `WriteLine(string)`, I would either call it `WriteLine` like before.
If we want this to be specific, I would change the  delegate to take `string` and `Exception` and move the formatting of the string to the DeveloperExperience part.